From 92c374b7ab60d6c78cca0ccc8055aac5958d9c01 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Dec 2024 20:45:43 +0800
Subject: [PATCH 158/163] AOSCOS: MIPS: loongson64_cpuautoplug: use
 {add,remove}_cpu() to replace cpu_{up,down}()

With commit 93ef1429e556 ("cpu/hotplug: Add new {add,remove}_cpu()
functions"), the "userspace safe" add_cpu() and remove_cpu() were
introduced to replace cpu_up() and cpu_down(), respectively. The latter
two functions were then hidden since commit 33c3736ec888 ("cpu/hotplug:
Hide cpu_up/down()"), causing build failure.

Use the safer add_cpu() and remove_cpu() instead.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/mips/loongson64/loongson64_cpuautoplug.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/loongson64/loongson64_cpuautoplug.c b/arch/mips/loongson64/loongson64_cpuautoplug.c
index a69a9ef9b11e..eb129404379a 100644
--- a/arch/mips/loongson64/loongson64_cpuautoplug.c
+++ b/arch/mips/loongson64/loongson64_cpuautoplug.c
@@ -265,7 +265,7 @@ static void increase_cores(int cur_cpus)
 
 	target_cpu = cpumask_next_zero(0, cpu_online_mask);
 	lock_device_hotplug();
-	cpu_up(target_cpu);
+	add_cpu(target_cpu);
 	get_cpu_device(target_cpu)->offline = false;
 	unlock_device_hotplug();
 }
@@ -280,7 +280,7 @@ static void decrease_cores(int cur_cpus)
 
 	target_cpu = find_last_bit(cpumask_bits(cpu_online_mask), num_possible_cpus());
 	lock_device_hotplug();
-	cpu_down(target_cpu);
+	remove_cpu(target_cpu);
 	get_cpu_device(target_cpu)->offline = true;
 	unlock_device_hotplug();
 }
-- 
2.47.1

